### DISCUSSION TITLE
No distance passed to HantushWellModel, assuming r=1.0.

### DISCUSSION THREAD

**Discussion by VeloVolant** [pastas/pastas#601] [Q&A] [answered]
I have seen this alert message "INFO: No distance passed to HantushWellModel, assuming r=1.0." in both a model I am currently making, and in the "adding multiple wells" example (strangely). Obviously, I would prefer that the model doesn't consider a distance of 1 instead of the real distances (e.g. 100 to 2000 meters). Unfortunately I don't understand how to get read of the problem, as the example itself seems to face it. Thks in advance for your help

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @VeloVolant, the short answer is that this message is not something you need to be worried about.

But be aware that if you're interested in the shape of the step or block response for a particular well that you need to explicitly calculate it for that well:

```python
# given some pastas Model ml
ml = ps.Model(...)

# and some WellModel
wm = ps.WellModel(...)
ml.add_stressmodel(wm)
ml.solve()

step_1m = ml.get_step_response("wells")  # <- this will return the step response for r=1 (hence the message)

step_well0 = wm.rfunc.step(wm.get_parameters(istress=0))  # <- this will get the response function for the first well

step_well1 = wm.rfunc.step(wm.get_parameters(istress=1))  # <- this will get the response function for the second well
```

---
The long answer, explaining why you're seeing this message:

When using `HantushWellModel()` with multiple wells you actually have multiple response functions, each one scaled by the distance parameter. When you use `pastas.Model` methods to get the shape of the response, e.g. `ml.get_step_response()`, it gets the parameters for the `HantushWellModel` response and passes those to the function that calculates the shape of the response. The `pastas.Model` object does not "know" about the different distances to different wells, because the distance is a known value, and not a parameter that needs to be optimized. So it only passes the three optimized parameters (`A`, `a`, and `b`, which does not include the distance `r`) to `HantushWellModel.step()` to calculate the response. This means `HantushWellModel` must make some assumption for the value of the distance `r`. By default this is `r=1.0`, which is the INFO message you're seeing. So you get the shape of the response function for a pumping well 1 meter away from your observation well. You're seeing this message twice because when calculating the step, it also needs to calculate the total duration of the response (`HantushWellModel.get_tmax()`), which is once again dependent on the distance parameter, causing an extra log message to appear. 

Since plotting the model results includes the step response, this confusing INFO log message appears (2x). 

---
If you've gotten this far, I agree that this message is confusing, and especially since it appears twice when doing standard stuff such as plotting the model results. 

Ideally we'd want to inform users that the step response is calculated for `r=1` when calling `ml.get_step_response()` but maybe stay silent when calling `ml.plots.results()`. I just haven't yet figured out a clean way to make that distinction. Maybe worth another try though...

**Reply by dbrakenhoff** [reply]
Just rememberd we discussed this earlier as well, in https://github.com/pastas/pastas/discussions/437 .

**Reply by VeloVolant** [reply]
Sorry for missing it, and many thanks for your clarifications anyway. VV
